### ISSUE TITLE
Error updating service endpoint in Azure DevOps: TF400898

### ISSUE THREAD

**Issue by alex-3sr** [microsoft/terraform-provider-azuredevops#427]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and Azure DevOps Provider) Version
```
Terraform v1.0.0
on windows_amd64
+ provider registry.terraform.io/dhensby/azuredevops v0.1.0
+ provider registry.terraform.io/hashicorp/azuread v1.6.0
+ provider registry.terraform.io/hashicorp/azurerm v2.68.0
+ provider registry.terraform.io/hashicorp/local v2.1.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/microsoft/azuredevops v0.1.6
```
### Affected Resource(s)

* `azuredevops_serviceendpoint_azurerm`

### Terraform Configuration Files

```hcl
resource "azuredevops_serviceendpoint_azurerm" "devops_endpoint_arm_agents" {
  project_id                = azuredevops_project.devops_projects.id
  service_endpoint_name     = "${local.project}-Deploy-${var.environment}"
  description               = "Build Agents DevOps"
  azurerm_spn_tenantid      = data.azurerm_client_config.current.tenant_id
  azurerm_subscription_id   = data.azurerm_subscription.current.subscription_id
  azurerm_subscription_name = data.azurerm_subscription.current.display_name

  credentials {
    serviceprincipalid  = azuread_application.azuread_app.application_id
    serviceprincipalkey = random_password.azuread_app_secretrandom.result
  }
}
```

### Expected Behavior
There are no change on this service endpoint, so no need to try update it

### Actual Behavior
During plan, Terraform want update the ressource but doesn't give the update who is need ->
```
  # azuredevops_serviceendpoint_azurerm.devops_endpoint_arm_agents will be updated in-place
  ~ resource "azuredevops_serviceendpoint_azurerm" "devops_endpoint_arm_agents" {
        id                        = "xxxx-eecd-467d-850b-yyyy"
        # (6 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }
```

I've this issue on multiple service endpoints, but not all. And the problem is especially like this is clearly not explain the update that terraform want to change, so i've no workaround for ignore_change temporary and permit to have green release.

Don't hesitate if need more debug, more information. I laready to to remove from state and import again, still have issue :(

Thanks
Alex

**Comment by xuzhang3** [comment]
Hi @alex-3sr I cannot reproduce your error. In your configurations, you import two ADO providesr: `provider registry.terraform.io/dhensby/azuredevops v0.1.0` and `provider registry.terraform.io/microsoft/azuredevops v0.1.6`. Can you remove one of them and try again?

**Comment by alex-3sr** [comment]
Hi,

I see that, you right, but it's not used anymore and I don't found anyway to remove it.
it was initialy the provider where we begin to use TF with AzDo, then after microsoft official provider we migrate to it.
I chechekd on the tfstate, I've no associated ressources with this old provider and on .terraform plugin folder, I don't have too.
In more, when i run **_terraform providers_** I got this ->
```
Providers required by configuration:
.
├── provider[registry.terraform.io/hashicorp/azuread] ~> 1.0
├── provider[registry.terraform.io/hashicorp/azurerm] ~> 2.0
├── provider[registry.terraform.io/hashicorp/random] ~> 3.0
├── provider[registry.terraform.io/hashicorp/local] ~> 2.0
└── provider[registry.terraform.io/microsoft/azuredevops] ~> 0.0

Providers required by state:

    provider[registry.terraform.io/hashicorp/azuread]

    provider[registry.terraform.io/hashicorp/azurerm]

    provider[registry.terraform.io/hashicorp/random]

    provider[registry.terraform.io/microsoft/azuredevops]
```

I will try to find how I can remove it, and will restart again a plan/apply for check if i've some change

Thx
Alex

**Comment by alex-3sr** [comment]
OK, so finally was more easy than expected :) I just find that .terraform.lock.hcl still reference old providers. Removed it, and now it's ok ->
```
Terraform v1.0.0
on windows_amd64
+ provider registry.terraform.io/hashicorp/azuread v1.6.0
+ provider registry.terraform.io/hashicorp/azurerm v2.68.0
+ provider registry.terraform.io/hashicorp/local v2.1.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/microsoft/azuredevops v0.1.6

Your version of Terraform is out of date! The latest version
is 1.0.3. You can update by downloading from https://www.terraform.io/downloads.html
```

So, i start again a plan like now I've only 1 version. Like explained, i've this detected updated, but on detail nothing whos me what updated is drifted
![image](https://user-images.githubusercontent.com/38663389/126830031-4bd40358-620e-45d2-9ee9-6d937e6d5ae8.png)

And on the apply stage, Still have same error :(
![image](https://user-images.githubusercontent.com/38663389/126830124-58a76b15-78ce-4cd2-a18f-a84ca4212928.png)

If need, I can add verbose logs if it can help

Thanks and have a nice WE
Alex

**Comment by tmeckel** [comment]
@alex-3sr Unfortunately `TF400898` is a general error which does not provide very much information what really happened.

You can list the current service connections and their configuration for a project by using the following REST endpoint:

`https://dev.azure.com/{organization}/{project}/_apis/serviceendpoint/endpoints?includeFailed=true&includeDetails=true&api-version=6.1-preview.4`

And if you compare the data from the REST endpoint with the `terraform` state, can you spot the difference why the provider wants to update the service connection?

**Comment by alex-3sr** [comment]
Hi,

OK thanks thant help me a lot, and think now why I got this error.
The main difference that I found between tfstate and devops service endpoint, is the subscription name (azurerm_subscription_name).

And I remember that Az subscriptions was rename after tf import, and like first I generate dynamicaly the az subscription name with data source and secondly it's not possible anymore to change az subscription name on devops endpoints, so I don't have any options now :) I can destroy/recreate the service endpoint on devops and set correctly the new subscription name or more quick to ignore subscription name on lifecycle.

I will try to explore theses options, but I think that we can archive this request. Except if need to follow it for not permit update on ressource azuredevops_serviceendpoint_azurerm if subscriptionId/subscriptionName has been update on Terraform code, update will never work, it need a full replace (destroy/recreate) for my point of view (the GUI doesn't permit to do it, so i think that Devops api too but not test yet).

Thanks
ALexandre
